### ISSUE TITLE
802.3ad bonding mode doesn't work on production server

### ISSUE THREAD

**Issue by galuha** [Mellanox/libvma#49]
Greetings

**Comment by rosenbaumalex** [comment]
we'll have to take a deeper look to check this and update you.
thanks for reporting.

**Comment by MohammadQurt** [comment]
Hi,
Regarding the issue you found, we still didn’t succeed to reproduce the issue.
According to the description and logs you attached we did the following to reproduce the issue (Please update us if we missed anything):

·         Server-Client Configurations:
1.       Used OFED 3.2-2.0.0.0 with ConnectX3 on both server and client.
2.       Configured the server with vlan over 802.3ad bonded interface.
3.       Set the server and client on a different networks.
4.       Configured a server to behave as a router to route packets between server and client.

·         802.3ad Bonding configurations (Tried Immediate and Permeant configurations with the following parameters combinations):
1.       mode=4 miimon=100
2.       mode=4 miimon=100 fail_over_mac=0
3.       mode=4 miimon=100 fail_over_mac=1

·         Server-Client Commands:
1.       Server: Sockperf TCP, 1 socket, non-blocking.
2.       Client: Sockperf TCP, Ping-Pong, 1 socket (non-blocking or blocking). 

Also, please can you update us with the following:
1.       Seems both logs belong to a server, so can you attach VMA_TRACELEVEL=4 logs for client side.
2.       Can you please run sysinfo (https://mellanox.my.salesforce.com/sfc/p/#500000007heg/a/50000000Xab4/VneS.zpLith9.GWZ.XthrStGzhuRBH9SZS_DmBENbfI) for every VMA machines and send us the output.
3.       You said: **It fails when server (vlan over bond)<-> router 802.3ad configured<-> internet <-> client. This works on kernel stack, hence net config is fine.**
What  are the commands you used for router 802.3 configurations.
4.       Is the issue reproducible every time?
Thanks in advance

**Comment by galuha** [comment]
Hello, here is the sysinfo file (remove .zip from behind):
[sysinfo-snapshot-v3.1.7-ofed-20160504-1945.tgz.zip](https://github.com/Mellanox/libvma/files/249216/sysinfo-snapshot-v3.1.7-ofed-20160504-1945.tgz.zip)

I will be able to get client side log later.

· Server-Client Configurations:
1. Used OFED 3.2-2.0.0.0 with ConnectX3 on both server and client.
2. Configured the server with vlan over 802.3ad bonded interface.
3. Set the server and client on a different networks.
4. Configured a server to behave as a router to route packets between server and client. - it's not correct. I have an actual router between server and the client, configured LACP 802.3ad

What are the commands you used for router 802.3 configurations:
[network-scripts.zip](https://github.com/Mellanox/libvma/files/250731/network-scripts.zip)
route add default gw 172.25.24.1 device bond0.380

**Comment by MohammadQurt** [comment]
Hi galuha,

Thanks for the update.

Can you please retry with the latest vma version here on githup: https://github.com/Mellanox/libvma

Thanks

**Comment by mellanoxer** [comment]
I have the same issue. VMA7.0.14 does not work correctly for TCP connections if vlan over bonding is used. Without libvma TCP connection is succesful.

**Comment by mellanoxer** [comment]
I've checked VMA 8.0.3. TCP connect() does not work if using vlan over bond.
Created socket is marked as offloaded. But program cant go out of the connect() and frezes.

**Comment by mellanoxer** [comment]
galuha, please send me email to mellanoxer@yandex.ru, I've found solution

**Comment by mellanoxer** [comment]
When my client invokes connect() VMA sucessfully sends SYN to server (I see it in VMA-logs as fast_send() call). Server replied with SYN-ACK. But this packet does not achive vma "engine" of client and go to the OS (as I can see SYN-ACK in tcpdump). So client freezes when calling connect(), and doesn't receive any replies from server. There is obvious error in processing rx-rings inside VMA when using TCP and vlan over bond. The socket considered to be offloaded, bonding interface considered to be offloaded, but there is no packets receiving in VMA.
At the contrary, when using UDP on vlan over bond - it is worked. But not TCP.

**Comment by mellanoxer** [comment]
rosenbaumalex, please send me email to mellanoxer@yandex.ru, and I'll send back you my VMA-logs of client with VMA_TRACELEVEL=ALL. Unfortunatly, I can't public this log here (for security reasons).
It will help to close this issue. Thank you.

**Comment by mellanoxer** [comment]
Is there any hope to solve this issue? 
What kind of tests I should to perform for helping libvma developers?

**Comment by galuha** [comment]
I guess there is no hope unless devs can reproduce the problem.
MohammadQurt, I am offering the team viewer session to the server with the issue. Please let me know if you are willing to see it.

**Comment by mellanoxer** [comment]
May be you will add some additional log-lines in sources according to logs I previosly posted to you.
To get more details. I will run again with your patch and send you back more detailed logs.

**Comment by rosenbaumalex** [comment]
We're still having issues with reproducing this.
We're looking at the update log we received from your setup.
Still no root cause understanding.

PS: best is if you open a support@mellanox.com ticket to get full tracking of this incident

**Comment by mellanoxer** [comment]
support@mellanox.com never answer me and I dont think I will buy support subscription, because now I'm thinking about to refuse of using Mellanox in my servers and try Solar.. It's too much errors and comparably low reliability.

**Comment by OphirMunk** [comment]
@galuha We are not able to reproduce your case. We tried it by several engineers and a Field Application Engineer. 
Is your team viewer session still relevant? Please let know

**Comment by galuha** [comment]
Yes, it is. Skype me for more details. I just have sent my skype id to you via email

**Comment by OphirMunk** [comment]
@galuha FAE will return to you within 1-2 weeks
